src/Suppressor.jl

```diff
@@ -121,20 +136,24 @@ macro capture_out(block)
         if ccall(:jl_generating_output, Cint, ()) == 0
             original_stdout = stdout
             out_rd, out_wr = redirect_stdout()
-            out_reader = @async read(out_rd, String)
+            io_buf = IOBuffer()
+            out_reader = @async _safe_write_ignore_close_task(io_buf, out_rd)
         end
 
         try
             $(esc(block))
         finally
             if ccall(:jl_generating_output, Cint, ()) == 0
                 redirect_stdout(original_stdout)
-                close(out_wr)
+                flush(out_wr)
             end
         end
 
         if ccall(:jl_generating_output, Cint, ()) == 0
-            fetch(out_reader)
+            s = String(take!(io_buf))
+            # Now close the task to allow the Pipe() to be GC'd
+            close(io_buf)
```

Comment:
I _think_ at this point you need write something to the pipe and flush to make sure the task dies.